bokehjs/src/coffee/common/embed.coffee

```diff
@@ -161,6 +164,17 @@ embed_items = (docs_json, render_items, websocket_url) ->
           console.log("Error rendering Bokeh items ", error)
       )
 
+handle_notebook_comms = (msg) ->
+  data = JSON.parse(msg.content.data)
+  doc = documents[data.docid]
+  if not doc?
+    throw "handle_notebooks_comms received unknown document id: #{data.docid}"
+  source = doc.get_model_by_id(data.source.id)
+  if not source?
+    throw "handle_notebooks_comms could not locate data source for id: #{data.source.id}"
+  source.set('data', data.source.data)
+
```

Comment:
this could use the general create_json_patch / apply_json_patch methods on Document if we want, instead of hard coding ColumnDataSource.data here.